apps/staking/src/components/WalletButton/index.tsx

```diff
@@ -41,13 +46,18 @@ import {
   type States,
   useApi,
 } from "../../hooks/use-api";
+import { StateType as DataStateType, useData } from "../../hooks/use-data";
 import { useLogger } from "../../hooks/use-logger";
 import { usePrimaryDomain } from "../../hooks/use-primary-domain";
 import { AccountHistory } from "../AccountHistory";
 import { Button } from "../Button";
 import { ModalDialog } from "../ModalDialog";
 import { TruncatedKey } from "../TruncatedKey";
 
+const ONE_SECOND_IN_MS = 1000;
+const ONE_MINUTE_IN_MS = 60 * ONE_SECOND_IN_MS;
+const REFRESH_INTERVAL = 1 * ONE_MINUTE_IN_MS;
```

Comment:
@keyvankhademi it's a bit clunky because this is duplicated in two places since the dashboard data needs to be loaded now in the wallet menu too so we can display which account is the main account.  So we have to do the `useData` call on the dashboard data.  But it won't trigger a separate network call because swr will re-use the global cache, and this will go away when I refactor the dashboard data into use-api eventually, so I don't think it's a big issue to leave the duplication for now.

Reply:
oh ignore my last comment then, i just saw this.